mooncake-transfer-engine/include/transfer_engine_c.h

```diff
@@ -126,13 +128,14 @@ int unregisterLocalMemoryBatch(transfer_engine_t engine, void **addr_list,
 
 batch_id_t allocateBatchID(transfer_engine_t engine, size_t batch_size);
 
-int submitTransfer(transfer_engine_t engine, batch_id_t batch_id,
-                   struct transfer_request *entries, size_t count);
+mooncake::Status submitTransfer(transfer_engine_t engine, batch_id_t batch_id,
```

Comment:
seems we are exposing c++ class in pure c header?

Reply:
Changed to C style interface and adapted rust and go code accordingly.